support-frontend/assets/pages/[countryGroupId]/thank-you.tsx

```diff
@@ -105,30 +105,34 @@ export function ThankYou({ geoId, appConfig }: Props) {
 		return <div>Product not found</div>;
 	}
 
-	/** Get and validate ratePlan */
+	/** Get ratePlan */
 	const ratePlanParam = searchParams.get('ratePlan');
-	const ratePlanKey =
+	const ratePlanKeyOneOff =
+		productKey === 'Contribution' && !ratePlanParam ? 'OneOff' : undefined;
+	const ratePlanKeyRecurring =
 		ratePlanParam && ratePlanParam in product.ratePlans
 			? ratePlanParam
 			: undefined;
-	const ratePlan = ratePlanKey && product.ratePlans[ratePlanKey];
+	const ratePlanKey = ratePlanKeyOneOff ?? ratePlanKeyRecurring;
+	const ratePlan = ratePlanKey && (product.ratePlans[ratePlanKey] ?? 'OneOff');
```

Comment:
We should avoid hijacking this model to put `OneOff` in - as mentioned above, these are directly relating to the [Product API](https://product-catalog.guardianapis.com/product-catalog.json).

Reply:
I think there were hints on this approach in the original `thank-you.tsx` hence went down that path.  

Agree it's not ideal but seemed better approach than creating completely new thankyou page simple-copy just for `OneOff`. 

This PR can't be progressed without some sort of assumption on how to get to `OneOff`/`OneTime`/`1-Time`. Possibly I can finish the associated Storybook/Playwright test's (which will need) and leave it at that until our meeting decision - then revisit this PR (unless it needs to change drastically). Or it goes in and we revise afterwards?

Reply:
How about we have two different components that simply work out which thank you modules to show (one which knows about ratePlans etc and one which doesn't) - then both call a shared component which is responsible for rendering them?

Reply:
Will look at now. To confirm, do you mean two different thankyou components = two separate routes to each from their respective checkout? If so, currently I think as soon as routing is involved there is a requirement to have a (Zuora) product specified in the url however - would need to remove that I think.

Reply:
Yes I think we're on the same page. By routing I'm assuming you mean the scala server routes? You're right that expects a product so will have to change

Reply:
Ok, thats the part will ask George if can pair over, feels like we are lifting a fundamental routing product check however.